src/__experimental__/components/date/date-picker.component.js

```diff
@@ -109,17 +109,25 @@ function getDisabledDays(minDate, maxDate) {
     return null;
   }
 
-  if (minDate) {
+  if (minDate && checkIsoFormatLength(minDate)) {
     days.push({ before: DateHelper.stringToDate(minDate) });
   }
 
-  if (maxDate) {
+  if (maxDate && checkIsoFormatLength(maxDate)) {
     days.push({ after: DateHelper.stringToDate(maxDate) });
   }
 
   return days;
 }
 
+function checkIsoFormatLength(date) {
```

Comment:
This function is doing two things, checking the length and verifying the length of each section. I'm wondering if it should be split into two functions that do one thing each? Or at least in would be better renamed as `verifyIsoFormat()`.